### PR TITLE
if match case is null for querystring then use default value set to l…

### PR DESCRIPTION
**Pull request by sandiphudekar**
if match case is null for querystring then use default value set to layer metadata

### CODE REVIEW

File: mapwfs.c [hunk 1]
```diff
@@ -3376,6 +3377,7 @@ int msWFSGetFeature(mapObj *map, wfsParamsObj *paramsObj, cgiRequestObj *req,
   papszGMLGroups = (char**) calloc(sizeof(char*), map->numlayers);
   papszGMLIncludeItems = (char**) calloc(sizeof(char*), map->numlayers);
   papszGMLGeometries = (char**) calloc(sizeof(char*), map->numlayers);
+  papszWFSDefaultMatchCase = (char**)calloc(sizeof(char*), map->numlayers);
```
Comment (danleydmello) [inline mapwfs.c:3380]: I don't see this memory being freed anywhere. This should be free in erroneous situations as well.

**Review by danleydmello — CHANGES_REQUESTED** [review]
Most of the code looks good except the one with memory being not release

File: mapogcfilter.c [hunk 2]
```diff
@@ -742,7 +742,7 @@ int FLTLayerApplyPlainFilterToLayer(FilterEncodingNode *psNode, mapObj *map,
 /*      Calling function should use FreeFilterEncodingNode function     */
 /*      to free memeory.                                                */
 /************************************************************************/
-FilterEncodingNode *FLTParseFilterEncoding(const char *szXMLString)
+FilterEncodingNode *FLTParseFilterEncoding(const char *szXMLString,const char *defaultMatchCase)
```
Comment (simen-wu) [inline mapogcfilter.c:745]: Update the comment of this function since you add a new parameter.

File: mapogcfilter.c [hunk 3]
```diff
@@ -1473,6 +1478,12 @@ void FLTInsertElementInNode(FilterEncodingNode *psFilterNode,
           propIsLike->pszEscapeChar = msStrdup(pszEscapeChar);
 
           pszTmp = (char *)CPLGetXMLValue(psXMLNode, "matchCase", NULL);
+
+		  if (pszTmp == NULL)
```
Comment (simen-wu) [inline mapogcfilter.c]: Align this following block of code.

File: mapogcfilter.c [hunk 4]
```diff
@@ -1473,6 +1478,12 @@ void FLTInsertElementInNode(FilterEncodingNode *psFilterNode,
           propIsLike->pszEscapeChar = msStrdup(pszEscapeChar);
 
           pszTmp = (char *)CPLGetXMLValue(psXMLNode, "matchCase", NULL);
+
```
Comment (simen-wu) [inline mapogcfilter.c:1480]: Align this block of code. Also put the '{' on the same line of 'if' like others.

File: mapwfs.c [hunk 5]
```diff
@@ -4063,6 +4067,7 @@ int msWFSGetPropertyValue(mapObj *map, wfsParamsObj *paramsObj, cgiRequestObj *r
   char* pszGMLGroups = NULL;
   char* pszGMLIncludeItems = NULL;
   char* pszGMLGeometries = NULL;
+  char* papszWFSDefaultMatchCase = NULL;
```
Comment (simen-wu) [inline mapwfs.c]: Use name papszWFSDefaultMatchCases so it looks the same as others (e.g. pszGMLGroups, pszGMLIncludeItems etc.)

**Review by simen-wu — COMMENTED** [review]
Looks good. Just a few minor things regarding the code formatting and the memory issue as Danley mentioned.

**Review by danleydmello — APPROVED** [review]
Looks good

**Review by simen-wu — APPROVED** [review]
looks good.

**Pull request merged**